### PR TITLE
rework achievements

### PR DESCRIPTION
**Pull request by atappy**
this PR is mainly about reworking the achievements model, by adding some fields to the Achievement enum (the category and the milestone) and the way they are displayed (we now display only the dev and verfied badge in the profileList, and have specific fields for each achievement on profileActivity)

### CODE REVIEW

File: app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt [hunk 1]
```diff
@@ -55,20 +55,22 @@ open class FirebaseProfileService @Inject constructor(
                 db.collection(PROFILE_COLL).document(user.uuid),
                 FOLLOWING_PROFILES, FieldValue.arrayUnion(targetId)
             )
-            if (updateAchievementsFollower(user) != null) {
+            val updatedUser = user.copy(following = user.following.plus("newUser"))
```
Comment (TheblueMan003) [inline app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt:58]: Why is there an hardcorded "newUser" string here?

**Review by TheblueMan003 — APPROVED** [review]
Otherwise Looks good

File: app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt (hunk 1 above)
Comment (atappy) [reply app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt:58]: that is meant to add a +1 to the size of the follower/following list, in order to calculate achievements properly without re-fetching the user

**Review by TruddyTheDuddi — APPROVED** [review]
Seems good to me! The only thing that I would change are the hardcoded strings for the Achievements/Badges, so that it would be automatically translated in different languages

**Pull request merged**